text/0000-interactive-app-creation.md

```diff
@@ -0,0 +1,142 @@
+
+Evaluating criteria: 
+1. Is it hard (or time-consuming) to change it later? *For example, it's easy to remove the `ember-welcome-page` addon, so we did not include it. But it's time consuming to change from `yarn` to `npm` (or vice-versa) so we made sure to include that.* 
+1. Is it difficult to discover the option?
+1. What is the maintenance cost? *For example, a nice question to add could be something like, Do you want to use a CSS preprocessor? But, because Ember does not deal with SASS, LESS... by default, the interactive workflow will have to install an addon (like `ember-cli-less`) which would imply that the CLI team will have to ensure it never fails.* 
+1. Does the question affect another one? *For example, the LTS question can affect the CI file generated by the CI question (setup tests on CI with `ember-try`). Questions that modify the result of other questions too much should be avoided to prevent high maintenance.*
+
+### Triggering the Workflow
+
+The following commands, and reasonable combinations thereof, should work to enter into the interactive workflow: 
+
+1. `ember new`
+1. `ember new --interactive`
+1. `ember new -i`
+1. `ember new my-app -i`
+
+### App/Addon 
+
+Starting a new Ember app or addon as we do now (at the time of this writing) will not change. If a user types `ember new my-app` they will still get a new Ember app generated in the usual ways. However, if the user types `ember new` they will enter into an interactive workflow that will ask them a few key questions about their application. 
+
+If a user were to type in existing flags that are also questions in the interactive workflow (i.e., `ember new my-app --lang en-US --interactive`), it will skip the question entirely. 
+
+When a user types `ember new` into their command line, they will be asked these questions (defaults indicated by a `>`):
+
+- `Is this an app or an addon? (use arrow keys)`
+  - `> app`
+  - `addon`
+- `Please provide the name of your app/addon:`
+- `Please provide the spoken/content language of your app/addon: (use arrow keys)`
+  - `> en-US`
+  - `my computer's default language`
+  - `manually define a different language`
+- `Pick the package manager to use when installing dependencies:  (use arrow keys)`
+  - `> NPM`
+  - `Yarn`
+  - `ignore/skip`
+- `What CI system do you want to use? (use arrow keys)`
+  - `> travis-ci`
+  - `ignore/skip`
+
+#### App/Addon Name
+There will be no default for this question. The user must enter a response. 
+
+#### App/Addon Language
+
+- the default would be `en-US`
+- if your system has a different language set (which can be confirmed by using the `echo $LANG` command in the terminal), then _that_ language would be shown as the second option
+- if your system was already set to the default language, the second option would not be shown. 
+- for "manually define a language", we will validate the response against the allowed language codes
```

Comment:
Should we be responsible for this? I relate this to something like timezones, it's really hard to get right. Are we sure we wouldn't be excluding anyone, like maybe an obsure or brand code is introduced that we don't support? I'm don't know how static the list is.

Reply:
@kellyselden we're already doing this with the `--lang` flag. You're right that it could be a tricky issue but we have a fairly high level of confidence in the library that we're using for this. [(is-language-code)](https://www.npmjs.com/package/is-language-code).

> Are we sure we wouldn't be excluding anyone, like maybe an obscure or brand code is introduced that we don't support?

While new regions can be created (due to things like changes in political landscape), it's unlikely that there will be a new language code. All languages have codes, and even languages that no longer have living native speakers still have language codes. For example, `htc` is the code for the [Hattic](https://en.wikipedia.org/wiki/Hattic_language) language, a historical language from 4,000 years ago.

Reply:
I'm not so sure about this. There is a page on Wikipedia about this, but it may not be up to date: https://en.wikipedia.org/wiki/Category:Languages_without_ISO_639-3_code

However, using a continually updated library is probably good enough since you can always edit index.html afterward.